tpchgen-cli/src/main.rs

```diff
@@ -355,88 +299,63 @@ impl Cli {
-                Table::Part => self.generate_part().await?,
-                Table::Supplier => self.generate_supplier().await?,
-                Table::Partsupp => self.generate_partsupp().await?,
-                Table::Customer => self.generate_customer().await?,
-                Table::Orders => self.generate_orders().await?,
-                Table::Lineitem => self.generate_lineitem().await?,
-            }
+        // Determine what files to generate
+        let mut output_partition_plans = vec![];
+        for table in &tables {
+            self.create_plans(*table, &mut output_partition_plans)?;
         }
 
+        // force the creation of the distributions and text pool to so it doesn't
+        // get charged to the first table
+        let start = Instant::now();
+        debug!("Creating distributions and text pool");
+        Distributions::static_default();
+        TextPool::get_or_init_default();
+        let elapsed = start.elapsed();
+        info!("Created static distributions and text pools in {elapsed:?}");
+
+        // Run
+        let runner = runner::PlanRunner::new(output_partition_plans, self.num_threads);
+        runner.run().await?;
         info!("Generation complete!");
         Ok(())
     }
 
-    define_generate!(
-        generate_nation,
-        Table::Nation,
-        NationGenerator,
-        NationTblSource,
-        NationCsvSource,
-        NationArrow
-    );
-    define_generate!(
-        generate_region,
-        Table::Region,
-        RegionGenerator,
-        RegionTblSource,
-        RegionCsvSource,
-        RegionArrow
-    );
-    define_generate!(
-        generate_part,
-        Table::Part,
-        PartGenerator,
-        PartTblSource,
-        PartCsvSource,
-        PartArrow
-    );
-    define_generate!(
-        generate_supplier,
-        Table::Supplier,
-        SupplierGenerator,
-        SupplierTblSource,
-        SupplierCsvSource,
-        SupplierArrow
-    );
-    define_generate!(
-        generate_partsupp,
-        Table::Partsupp,
-        PartSuppGenerator,
-        PartSuppTblSource,
-        PartSuppCsvSource,
-        PartSuppArrow
-    );
-    define_generate!(
-        generate_customer,
-        Table::Customer,
-        CustomerGenerator,
-        CustomerTblSource,
-        CustomerCsvSource,
-        CustomerArrow
-    );
-    define_generate!(
-        generate_orders,
-        Table::Orders,
-        OrderGenerator,
-        OrderTblSource,
-        OrderCsvSource,
-        OrderArrow
-    );
-    define_generate!(
-        generate_lineitem,
-        Table::Lineitem,
-        LineItemGenerator,
-        LineItemTblSource,
-        LineItemCsvSource,
-        LineItemArrow
-    );
+    /// Appends  OutputPartitionPlan(s) for the given table to `plans`
+    fn create_plans(
+        &self,
+        table: Table,
+        plans: &mut Vec<OutputPartitionPlan>,
+    ) -> Result<(), io::Error> {
+        let generation_plan = GenerationPlan::try_new(
```

Comment:
My idea is to change this code to create multiple GenerationPlans for each file, potentially.